src/TypeResolver.js

```diff
@@ -127,15 +140,24 @@ class TypeResolver {
         }
 
         if (Array.isArray(valueTypes)) {
-            if (key !== 'record' && key !== 'variant') {
-                resolvedTypes = valueTypes.map(t => this.resolveType(t))
+            if (key !== 'variant') {
+                resolvedTypes = valueTypes.map(v => {
+                    const tpl = v.hasOwnProperty('type') ? v.type : v
+                    const t = vars.hasOwnProperty(tpl) ? vars[tpl] : tpl
+
+                    return this.resolveType(t, vars)
+                })
             }
         }
 
         if (key === 'void') {
             return FateTypeVoid()
         }
 
+        if (key === 'unit') {
+            return FateTypeTuple([])
```

Comment:
As I understand it is more native in JS to map `unit` type to `undefined` instead of the empty array.
This seems to be a similar issue to https://github.com/aeternity/aepp-calldata-js/issues/73

Reply:
This is the TypeResolver, `unit` is type alias to empty tuple and must be serialized to it. In Sophia and thus in the ACI, there is no specific `unit` type in ACI return type that can be used.

That is, this lib cannot return `undefined` in case of `unit` used somewhere in the code as this type is not in the ACI of return types.

Or you expect to be able to pass `undefined` as `unit` instead of the notation for empty tuple `[]` ?